src/styles/MuiThemeProvider.js

```diff
@@ -1,20 +1,29 @@
 import {Component, PropTypes} from 'react';
 import getMuiTheme from './getMuiTheme';
+import getUniqueIdGenerator from '../utils/getUniqueIdGenerator';
+
 
 class MuiThemeProvider extends Component {
 
   static propTypes = {
     children: PropTypes.element,
     muiTheme: PropTypes.object,
+    uniqueIdGen: PropTypes.func,
   };
 
   static childContextTypes = {
     muiTheme: PropTypes.object.isRequired,
+    uniqueIdGen: PropTypes.func.isRequired,
+  };
+
+  state = {
+    uniqueIdGen: this.props.uniqueIdGen || getUniqueIdGenerator(),
```

Comment:
Updating the `uniqueIdGen` may cause some issues. Mounted components will keep their ids `mui-id-0`, `mui-id-1`...But if there are components will mount, their `uniqueIdGen` may start at `0`, and their ids would be `mui-id-0`, `mui-id-1`...😰